src/js/wrap.h

```diff
@@ -238,22 +247,28 @@ namespace ccf::js
   class Context
   {
     JSContext* ctx;
+    JSRuntime* js_run_time;
 
   public:
     const TxAccess access;
+    UntrustedHostTime* host_time;
```

Comment:
Why is this a raw pointer?

Reply:
This needs to be a pointer. If we simply pass an address instead of a pointer, it will end up being a local variable and will go out of scope in the `Context::call` function

Reply:
We need to pass an address that will still be alive when the interrupt handler is called, but that doesn't need to be a raw pointer. We could use a `std::unique_ptr` to avoid remembering to manually call `delete`, or just use a direct struct member here (`UntrustedHostTime host_time;` rather than `UntrustedHostTime*;`). It'll be default constructed when `Context` is constructed, and automatically deleted in `~Context()`. When calling `JS_SetInterruptHandler`, we pass `&host_time`, and know that its lifetime spans the time where the interrupt handler could be called.

Reply:
> just use a direct struct member here (`UntrustedHostTime host_time;` rather than `UntrustedHostTime*;`). 

It's data will be assigned in `Context::call` and if it's a struct member instead of a pointer, that assignment will not work. (Because we can't assign class member variables in a member function).

> We could use a std::unique_ptr to avoid remembering to manually call delete

`std::unique_ptr` can't be copied into the `void *`. Maybe I could use `std::shared_ptr<void>` type. I will check that.

Reply:
> `std::unique_ptr` can't be copied into the `void *`. Maybe I could use `std::shared_ptr<void>` type. I will check that.

If `host_time` is declared as `std::unique_ptr<UntrustedHostTime> host_time;`, then we can call `host_time.get()` to return the raw `UntrustedHostTime*` (convertible to `void*`, `JS_SetInterruptHandler(..., host_time.get())` should work).

> It's data will be assigned in `Context::call` and if it's a struct member instead of a pointer, that assignment will not work. (Because we can't assign class member variables in a member function).

I don't follow this bit. If this was `UntrustedHostTime host_time;` (NB - non-const), then `host_time.start_time = curr_time;` in `Context::call(...)` should work fine?